github/docs
# Be more precise when expression syntax ${{ }} can be omitted in if statement

### Why:

Unfortunately, you can't always omit `${{ }}` in the `if` statement and it's not documented anywhere. Consider below examples

```yml
name: test-expression

on:
  push:
    branches:
      - main

jobs:
    test:
        runs-on: ubuntu-latest
        steps:
        - run: echo 'Hello, world!'
    statement-if:
        runs-on: ubuntu-latest
        if: github.event_name == 'push' && !startsWith(github.ref, 'refs/heads/')
        steps:
        - run: echo 'Summer!'
```

the above works as expected, but the below that only differs by order of statements in `if` doesn't work

```yml
name: test-expression

on:
  push:
    branches:
      - main

jobs:
    test:
        runs-on: ubuntu-latest
        steps:
        - run: echo 'Hello, world!'
    statement-if:
        runs-on: ubuntu-latest
        if: !startsWith(github.ref, 'refs/heads/') && github.event_name == 'push'
        steps:
        - run: echo 'Summer!'
```
ends with

<img width="443" alt="image" src="https://github.com/github/docs/assets/9593424/46e4d60a-2bd4-441e-81bf-0bc54c628056">

This is due to the YAML format see

- https://github.com/orgs/community/discussions/26303#discussioncomment-6739171
- https://github.com/orgs/community/discussions/26386#discussioncomment-3251729

basically, it can't start with `!`, because it has meaning in YAML specification.

> In my opinion, it's a pain to have it like that and it should be forbidden at all or work as expected even at the cost of breaking a little bit of specification of YAML. It should be reported to the team responsible for GitHub Workflows to take some action. 





### What's being changed (if available, include any code snippets, screenshots, or gifs):

The description for the `if` conditional.



### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline.

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).
